wdl/ExpansionHunter.wdl

```diff
@@ -142,3 +165,145 @@ task RunExpansionHunter {
+        else
+            "python:slim-buster"
+
+    command <<<
+        set -euxo pipefail
+
+        python <<CODE
+        import json
+        import sys
+        import os
+        from pathlib import Path
+
+        filename = "~{variant_catalog}"
+        filename_without_ext = Path(filename).stem
+        output_prefix = "~{output_prefix_}"
+        i = 0
+        subset_counter = 0
+
+        def serialize():
+            print(f"subset counter: {subset_counter}")
+            with open(f"{output_prefix}{filename_without_ext}_{subset_counter}.json", "w") as f_out:
+                json.dump(subset_catalogs, f_out, indent=4)
+
+        with open(filename, "r") as f_in:
+            catalogs = json.load(f_in)
+
+            subset_catalogs = []
+            for catalog in catalogs:
+                subset_catalogs.append(catalog)
+                i += 1
+                if i == ~{batch_size}:
+                    serialize()
+                    i = 0
+                    subset_counter += 1
+                    subset_catalogs = []
+            serialize()
+        CODE
+    >>>
+
+    RuntimeAttr runtime_attr_str_profile_default = object {
+        cpu_cores: 1,
+        mem_gb: 4,
+        boot_disk_gb: 10,
+        preemptible_tries: 3,
+        max_retries: 1,
+        disk_gb: 10
+    }
+    RuntimeAttr runtime_attr = select_first([
+        runtime_attr_override,
+        runtime_attr_str_profile_default])
+
+    runtime {
+        docker: split_variant_catalog_docker_
+        cpu: runtime_attr.cpu_cores
+        memory: runtime_attr.mem_gb + " GiB"
+        disks: "local-disk " + runtime_attr.disk_gb + " HDD"
+        bootDiskSizeGb: runtime_attr.boot_disk_gb
+        preemptible: runtime_attr.preemptible_tries
+        maxRetries: runtime_attr.max_retries
+    }
+}
+
+task ConcatEHOutputs {
+    input {
+        Array[File] vcfs_gz
+        Array[File] jsons
+        Array[File] overlapping_reads
+        Array[File] timings
+        String? output_prefix
+        String expansion_hunter_docker
+        RuntimeAttr? runtime_attr_override
+    }
+
+    output {
+        File json = "${output_prefix}.json"
+        File vcf_gz = "${output_prefix}.vcf.gz"
+        File overlapping_reads = "${output_prefix}.bam"
+        File timing = "${output_prefix}_timing.tsv"
+    }
+
+    command <<<
+        set -euxo pipefail
+
+        jq -s 'reduce .[] as $item ({}; . * $item)' ~{sep=" " jsons} > ~{output_prefix}.json
+
+        VCFS="~{write_lines(vcfs_gz)}"
+        bcftools concat --no-version --naive --naive-force --output-type z --file-list ${VCFS} --output "~{output_prefix}.vcf.gz"
+
+        BAMS="~{write_lines(overlapping_reads)}"
+        samtools merge ~{output_prefix}.bam -b ${BAMS}
+
+        TIMINGS=(~{sep=" " timings})
+        FIRST_TIMING=${TIMINGS[0]}
+        head -1 ${FIRST_TIMING} > ~{output_prefix}_timing.tsv
+        tail -n +2 -q ~{sep=" " timings} >> ~{output_prefix}_timing.tsv
```

Comment:
I've found using `tail -n +N` to be inefficient (I think it has to load the whole output into memory). Are the header lines indicated somehow, i.e. beginning with `#`? If so I would rewrite like `cat ~{sep=" " timings} | grep -v ^# >> ~{output_prefix}_timing.tsv`

Reply:
That would be ideal if the header was starting with `#`. Do you think awk would be any faster/readable?!

Reply:
Yeah awk is better here